### PR TITLE
Add additional Linux AF_PACKET options

### PR DESCRIPTION
**Pull request by nathaniel-bennett**
Added socket options, structs and other miscellaneous constants that make memory-mapped packet sockets possible in Linux (PACKET_TX_RING, PACKET_RX_RING, etc.).

### CODE REVIEW

File: src/unix/linux_like/linux/mod.rs [hunk 1]
```diff
@@ -140,13 +148,135 @@ s! {
...
+        pub tp_sec: ::__u32,
+        pub tp_nsec: ::__u32,
+        pub tp_snaplen: ::__u32,
+        pub tp_len: ::__u32,
+        pub tp_status: ::__u32,
+        pub tp_mac: ::__u16,
+        pub tp_net: ::__u16,
+        #[cfg(libc_union)]
+        pub hv1: ::__c_anonymous_hdr_variant,
+        #[cfg(not(libc_union))]
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/mod.rs:259]: libc_union no longer exists and should be removed.

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks a lot! @bors r+

**Comment by nathaniel-bennett** [comment]
For some reason, musl would throw compiler errors because `struct fanout_args` was missing. I've added a check that skips this and a FIXME; I presume the issue is taht musl CI has old headers (mentioned in other seccomp structs as well).

**Comment by JohnTitor** [comment]
Yeah, we use musl 1.1.24. A PR to update it has been submitted but it's still WIP.
@bors r+

**Pull request merged**